types/node/index.d.ts

```diff
@@ -424,28 +446,59 @@ declare namespace NodeJS {
     type NewListenerListener = (type: string | symbol, listener: (...args: any[]) => void) => void;
     type RemoveListenerListener = (type: string | symbol, listener: (...args: any[]) => void) => void;
 
-    export interface Socket extends ReadWriteStream {
-        isTTY?: true;
+    export interface StdStream extends ReadWriteStream, Readable, Writable {
+        // fs.ReadStream, fs.WriteStream
+        close(): void;
+        path: string;
+
+        // net.Socket
+        connect(port: number, host?: string, connectionListener?: Function): void;
+        connect(path: string, connectionListener?: Function): void;
+        write(buffer: Buffer): boolean;
+        write(buffer: Buffer, cb?: Function): boolean;
+        write(str: string, cb?: Function): boolean;
+        write(str: string, encoding?: string, cb?: Function): boolean;
+        write(str: string, encoding?: string, fd?: string): boolean;
+        write(data: any, encoding?: string, callback?: Function): void;
+        setTimeout(timeout: number, callback?: Function): void;
+        setNoDelay(noDelay?: boolean): void;
+        setKeepAlive(enable?: boolean, initialDelay?: number): void;
+        address(): { port: number; family: string; address: string; };
+        unref(): void;
+        ref(): void;
+        bufferSize: number;
+        localAddress: string;
+        localPort: number;
+        bytesRead: number;
+        bytesWritten: number;
+        connecting: boolean;
+        destroyed: boolean;
+
+        // tty.ReadStream
+        isTTY: boolean;
     }
 
-    export interface ProcessEnv {
-        [key: string]: string | undefined;
+    export interface StdReadStream extends StdStream {
+        // tty.ReadStream
+        isRaw: boolean;
+        setRawMode(mode: boolean): void;
     }
 
-    export interface WriteStream extends Socket {
-        columns?: number;
-        rows?: number;
+    export interface StdWriteStream extends StdStream {
+        // tty.WriteStream
+        columns: number;
+        rows: number;
     }
-    export interface ReadStream extends Socket {
-        isRaw?: boolean;
-        setRawMode?(mode: boolean): void;
+
+    export interface ProcessEnv {
+        [key: string]: string | undefined;
     }
 
     export interface Process extends EventEmitter {
-        stdout: WriteStream;
-        stderr: WriteStream;
-        stdin: ReadStream;
-        openStdin(): Socket;
+        stdout: StdWriteStream;
+        stderr: StdWriteStream;
+        stdin: StdReadStream;
```

Comment:
The `StdReadStream` you defined has a `setRawMode()` and `isRaw` like `tty.ReadStream`.
If you pipe in a file to node `process.stdin` doesn't have them.